### PR TITLE
Adds placeholder value for Rich Text fields on mobile

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://github.com/Fliplet/fliplet-studio/issues/5293

## Description
Had added a placeholder on mobile devices

## Screenshots/screencasts
https://share.getcloudapp.com/bLumKm0A

## Backward compatibility

This change is fully backward compatible.

## Notes
Apparently mobile theme still doesn't support 3rd part plugins as it was mentioned on the TinyMCE official community.
https://community.tiny.cloud/communityQuestion?id=90661000000TOzvAAG
https://community.tiny.cloud/communityQuestion?id=9066100000097myAAA
To work around this issue I've added the code from [this plugin](https://github.com/baangfilip/tinymce-placeholder), with slide adjustment for our case, not like a plugin but like a simple code so TinyMCE could work with it both on mobile version and web.

### CODE REVIEW

**Comment by YaroslavOvdii** [comment]
## Codacy Errors: 

**Expected method shorthand.** - can't user shorthands because we don't have a WebPack in this repository and that is why we are using es5 syntax.

**Strings must use doublequote.** - duo the Airbnb style guide we are using a single quote.

**Comment by YaroslavOvdii** [comment]
@tonytlwu 
Added requested changes from this [comment](https://github.com/Fliplet/fliplet-widget-theme/pull/72#issuecomment-598707173).
New screencast https://share.getcloudapp.com/v1urg2Yg

**Comment by tonytlwu** [comment]
@YaroslavOvdii It seems better. I think your best test in your screencast would be to type `Placeholder` because you can test and confirm that the position for `P` is exactly the same.

**Comment by tonytlwu** [comment]
And looks like you started the most recent screencast too late. I don't get to see what the placeholder looked like before you open the WYSIWYG fulls creen editor

**Comment by YaroslavOvdii** [comment]
@tonytlwu Adjusted paddings for native devices and web.
Screencast https://share.getcloudapp.com/bLumKAoP

**Comment by tonytlwu** [comment]
@YaroslavOvdii If you switch between 0:22 and 0:30 you'll find that the fonts are different. The same if you switch between 0:01 and 0:37. The only thing that should be different is the color.

Also, could you test with a placeholder text that is long enough for multiple lines? This way we can confirm that the line height, padding etc. are the same too.

**Comment by YaroslavOvdii** [comment]
@tonytlwu 
I've changed the current solution to support the multiple line entrance.
How it looks now https://share.getcloudapp.com/qGuoZoXB

**Comment by tonytlwu** [comment]
@YaroslavOvdii Your video needs to show entry text that is exactly the same as the placeholder text. Both need to be multiline. I need to compare them to make sure they look exactly the same, and it's only colours that are different.

**Comment by YaroslavOvdii** [comment]
@tonytlwu I check it first and on mobile font-weight was different on the placeholder and normal text, I've fixed it, could you please take a look at it one more time?
https://share.getcloudapp.com/jkuKAlo1
Screenshots:
screenshot-2020-03-19_09 04 56 223
screenshot-2020-03-19_09 04 23 977

**Comment by tonytlwu** [comment]
@YaroslavOvdii The placeholder text and user input don't look the same in the fullscreen editor. You can see by switching between 0:08 and 0:15. I've also done a composite of the two states to demonstrate this.

image

Also, can your video include how it looks after you click on the ⬅️arrow so it could include how the user input looks outside of the fullscreen editor? I need to review them too.

The goal is simple. They need to look exactly the same. Padding, margin, line height, font family, font size, font style, font decoration and font weight. The only thing that can be different is the colour.

**Comment by YaroslavOvdii** [comment]
@tonytlwu Added new placeholder styles so it would look the same as a text we write
New screencast: https://share.getcloudapp.com/yAuv9D7D
New screenshots: 
screenshot-2020-03-20_12 22 51 504
screenshot-2020-03-20_12 23 07 221
screenshot-2020-03-20_12 23 14 753
screenshot-2020-03-20_12 23 18 399

File: js/components/wysiwyg.js [hunk 1]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+          lineHeight: '19px',
+          padding: tinyMCE.DOM.getStyle(contentAreaContainer, 'padding', true),
+          width:'98%',
+          overflow: 'hidden',
+          'white-space': 'pre-wrap',
+          'font-weight': 'normal',
+          'font-size': '16px'
+        }
+      };
+      var placeholder_attrs = this.editor.settings.placeholder_attrs || defaultStyles;
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]:
```suggestion
      var placeholderAttrs = this.editor.settings.placeholder_attrs || defaultStyles;
```

File: js/components/wysiwyg.js [hunk 2]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+
+      tinymce.DOM.setStyle(contentAreaContainer, 'position', 'relative');
+
+      // Create label element in the TinyMCE editor
+      this.labelElement = tinymce.DOM.add( contentAreaContainer, this.editor.settings.placeholder_tag || "p", placeholder_attrs, placeholder_text );
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]:
```suggestion
      this.labelElement = tinymce.DOM.add(contentAreaContainer, this.editor.settings.placeholder_tag || "p", placeholderAttrs, placeholderText);
```

File: js/components/wysiwyg.js [hunk 3]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+    },
+    placeholderLabelHide: function() {
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]:
```suggestion
    hidePlaceholderLabel: function() {
```

File: js/components/wysiwyg.js [hunk 4]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+      tinymce.DOM.setStyle( this.labelElement, 'display', 'none' );
+    },
+    placeholderLabelShow: function() {
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]:
```suggestion
    showPlaceholderLabel: function() {
```

File: js/components/wysiwyg.js [hunk 5]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+      };
+      var placeholder_attrs = this.editor.settings.placeholder_attrs || defaultStyles;
+
+      tinymce.DOM.setStyle(contentAreaContainer, 'position', 'relative');
+
+      // Create label element in the TinyMCE editor
+      this.labelElement = tinymce.DOM.add( contentAreaContainer, this.editor.settings.placeholder_tag || "p", placeholder_attrs, placeholder_text );
+    },
+    placeholderLabelHide: function() {
+      tinymce.DOM.setStyle( this.labelElement, 'display', 'none' );
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]:
```suggestion
      tinymce.DOM.setStyle(this.labelElement, 'display', 'none');
```

File: js/components/wysiwyg.js [hunk 6]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+    },
+    placeholderLabelShow: function() {
+      tinymce.DOM.setStyle( this.labelElement, 'display', '' );
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]:
```suggestion
      tinymce.DOM.setStyle(this.labelElement, 'display', '');
```

File: js/components/wysiwyg.js [hunk 7]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+    },
+    placeholderOnFocus: function() {
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]:
```suggestion
    onPlaceholderFocus: function() {
```

Comment (tonytlwu) [reply js/components/wysiwyg.js]: Change all other similar functions too

File: js/components/wysiwyg.js [hunk 8]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+      this.editor.execCommand('mceFocus', false);
+    },
+    placeholderOnBlur: function() {
+      if (!this.editor.getContent()) {
+        this.placeholderLabelShow();
+      } else {
+        this.placeholderLabelHide();
+      }
+    },
+    addPlaceHolder: function() {
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]: Placeholder is one word.

```suggestion
    addPlaceholder: function() {
```

File: js/components/wysiwyg.js [hunk 9]
```diff
@@ -47,9 +47,67 @@ Fliplet.FormBuilder.field('wysiwyg', {
...
+      tinymce.DOM.bind(this.labelElement, 'click', this.placeholderOnFocus);
+      this.editor.on('focus', this.placeholderOnFocus);
+      this.editor.on('blur', this.placeholderOnBlur);
+      this.editor.on('change', this.placeholderOnBlur);
+      this.editor.on('setContent', this.placeholderOnBlur);
+      this.editor.on('keydown', this.placeholderLabelHide);
     }
   },
-  mounted: function () {
+  mounted: function() {
```
Comment (tonytlwu) [inline js/components/wysiwyg.js]: Anonymouse functions always have a space between `function` and `()`. Check them all.

```suggestion
  mounted: function () {
```

**Comment by YaroslavOvdii** [comment]
@tonytlwu fixed

**Pull request merged**